Wikipedia/Code/WMFTableOfContentsPresentationController.swift

```diff
@@ -107,9 +128,32 @@ public class WMFTableOfContentsPresentationController: UIPresentationController
     
     override public func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator transitionCoordinator: UIViewControllerTransitionCoordinator) {
         super.viewWillTransitionToSize(size, withTransitionCoordinator: transitionCoordinator)
-        
+
         transitionCoordinator.animateAlongsideTransition({(context: UIViewControllerTransitionCoordinatorContext!) -> Void in
             self.backgroundView.frame = self.containerView!.bounds
             }, completion:nil)
+
+        if(UIDeviceOrientationIsLandscape(UIDevice.currentDevice().orientation))
```

Comment:
also FYI, device orientation != application/view orientation. the device can be in any orientation, but the application (or current view controller) can be in another based on many different variables.